doc/constraint_causal_discovery.rst

```diff
@@ -0,0 +1,132 @@
+.. _constraint_causal_discovery:
+
+==================================
+Constraint-based causal discovery
+==================================
+.. automodule:: dodiscover.constraint
+    :no-members:
+    :no-inherited-members:
+.. currentmodule:: dodiscover
+
+The following are a set of methods intended for (non-parametric) structure learning
+of causal graphs (i.e. causal discovery) given observational and/or interventional data
+by checking constraints in the form of conditional independences (CI). At a high level
+all constraint-based causal discovery, tests CI statements, where we use different CI
+statistical tests to test the following null hypothesis:
+
+:math:`H_0: X \perp Y | Z` and :math:`H_A: X \not\perp Y | Z`
+
+For a given node ``X`` and ``Y`` in the underlying causal graph of interest, and
+a conditioning set, ``Z``. CI tests can have a variety of different assumptions
+that make one better than another in different settings and data assumptions.
+For more information on the CI tests themselves, see :ref:`conditional_independence`.
+
+Fundamental Assumptions of Constraint-Based Causal Discovery
+------------------------------------------------------------
+The fundamental assumptions of all algorithms in this section are the Markov property assumption
+and the causal faithfulness assumption :footcite:`Spirtes1993` and :footcite:`Pearl_causality_2009`.
+
+The Markov assumption states that all d-separation statements in the causal graph imply a
+corresponding CI statement in the data. This is a core-assumption that users from graphical modeling
+may be familiar with.
+
+On the other hand, the causal faithfulness assumption states that all
```

Comment:
I think I read somewhere that the chances of a faithfulness violation has measure zero when there are no latents and no deterministic relationships between variables.  Do we want to say that faithfulness violations are not that big a deal relative to violations of other assumptions

Reply:
Sure that is fair, but for dealing with real data, the problem of "weak-faithfulness" is highly likely. I can add some discussion around that.

ref: https://www.jstor.org/stable/23566569?casa_token=kzePz4xv42MAAAAA:PuhX4bXVGqes-QdqLTysC1wTY_gie2IwDRda1TACReQsu4g8mh3ewb_Hw_4mu7mzP3L8G_PW-Cgf7SMuJYL2UIwANUcVu3xVgGweU6_k0X9GjtX4LgE

Reply:
I agree with Adam, although faithfulness is very unlikely in practice, a weak form of it would already be an issue for the tests. And that can be quite common.